src/simulators/circuit_executor.hpp

```diff
@@ -204,6 +215,8 @@ template <class state_t>
 Executor<state_t>::Executor() {
   max_memory_mb_ = 0;
   max_gpu_memory_mb_ = 0;
+  required_memory_mb_ = 0;
+  set_required_memory_mb_ = false;
```

Comment:
I know that calling `required_memory()` multiple time is inefficient. However, `CircuitExecutor` should be stateless about `Circuit` and `Config` to be simulated. If we have these cache, `circuit` and `config` need to be `const` members of `CircuitExecutor`.

Reply:
Now required_memory_mb is ccalled once in controller, but called several times in executor to make it stateless